internal/status/httproute_test.go

```diff
@@ -77,3 +87,92 @@ func TestPrepareHTTPRouteStatus(t *testing.T) {
 		t.Errorf("prepareHTTPRouteStatus() mismatch (-want +got):\n%s", diff)
 	}
 }
+
+func TestConvertRouteConditions(t *testing.T) {
+	routeConds := []conditions.RouteCondition{
+		{
+			Type:    "Test",
+			Status:  metav1.ConditionTrue,
+			Reason:  "reason1",
+			Message: "message1",
+		},
+		{
+			Type:    "Test",
+			Status:  metav1.ConditionFalse,
+			Reason:  "reason2",
+			Message: "message2",
+		},
+	}
+
+	var generation int64 = 1
+	transitionTime := metav1.NewTime(time.Now())
+
+	expected := []metav1.Condition{
+		{
+			Type:               "Test",
+			Status:             metav1.ConditionTrue,
+			ObservedGeneration: generation,
+			LastTransitionTime: transitionTime,
+			Reason:             "reason1",
+			Message:            "message1",
+		},
+		{
+			Type:               "Test",
+			Status:             metav1.ConditionFalse,
+			ObservedGeneration: generation,
+			LastTransitionTime: transitionTime,
+			Reason:             "reason2",
+			Message:            "message2",
+		},
+	}
+
+	result := convertRouteConditions(routeConds, generation, transitionTime)
+	if diff := cmp.Diff(expected, result); diff != "" {
+		t.Errorf("convertRouteConditions() mismatch (-want +got):\n%s", diff)
+	}
+}
+
+func TestDeduplicateConditions(t *testing.T) {
+	g := NewGomegaWithT(t)
```

Comment:
I gave it a thought. Here I used gomega because `ConsistsOf` checks for equality without order. The suggested consistency makes sense to me. However, gomega output can be harder to parse.

For example, considering the output from gocmp:
```
=== RUN   TestPrepareHTTPRouteStatus
    httproute_test.go:85: prepareHTTPRouteStatus() mismatch (-want +got):
          v1beta1.HTTPRouteStatus{
          	RouteStatus: v1beta1.RouteStatus{
          		Parents: []v1beta1.RouteParentStatus{
          			{
          				ParentRef:      {Namespace: &"test", Name: "gateway", SectionName: &"accepted"},
          				ControllerName: "test.example.com",
          				Conditions: []v1.Condition{
          					{
          						Type:               "Accepted",
          						Status:             "True",
        - 						ObservedGeneration: 2,
        + 						ObservedGeneration: 1,
          						LastTransitionTime: {Time: s"2022-11-04 09:40:30.072751 -0400 EDT m=+0.018787859"},
          						Reason:             "",
          						Message:            "",
          					},
          				},
          			},
          			{ParentRef: {Namespace: &"test", Name: "gateway", SectionName: &"not-accepted"}, ControllerName: "test.example.com", Conditions: {{Type: "Accepted", Status: "False", ObservedGeneration: 1, LastTransitionTime: {Time: s"2022-11-04 09:40:30.072751 -0400 EDT m=+0.018787859"}, ...}}},
          		},
          	},
          }
--- FAIL: TestPrepareHTTPRouteStatus (0.00s)
```
```
g.Expect(result).Should(gomega.Equal(expected))
```

And the the output from Gomega:
```
=== RUN   TestPrepareHTTPRouteStatus
    httproute_test.go:90: 
        Expected
            <v1beta1.HTTPRouteStatus>: {
                RouteStatus: {
                    Parents: [
                        {
                            ParentRef: {
                                Group: nil,
                                Kind: nil,
                                Namespace: "test",
                                Name: "gateway",
                                SectionName: "accepted",
                                Port: nil,
                            },
                            ControllerName: "test.example.com",
                            Conditions: [
                                {
                                    Type: "Accepted",
                                    Status: "True",
                                    ObservedGeneration: 1,
                                    LastTransitionTime: {
                                        Time: 2022-11-04T09:44:40.528817-04:00,
                                    },
                                    Reason: "",
                                    Message: "",
                                },
                            ],
                        },
                        {
                            ParentRef: {
                                Group: nil,
                                Kind: nil,
                                Namespace: "test",
                                Name: "gateway",
                                SectionName: "not-accepted",
                                Port: nil,
                            },
                            ControllerName: "test.example.com",
                            Conditions: [
                                {
                                    Type: "Accepted",
                                    Status: "False",
                                    ObservedGeneration: 1,
                                    LastTransitionTime: {
                                        Time: 2022-11-04T09:44:40.528817-04:00,
                                    },
                                    Reason: "",
                                    Message: "",
                                },
                            ],
                        },
                    ],
                },
            }
        to equal
            <v1beta1.HTTPRouteStatus>: {
                RouteStatus: {
                    Parents: [
                        {
                            ParentRef: {
                                Group: nil,
                                Kind: nil,
                                Namespace: "test",
                                Name: "gateway",
                                SectionName: "accepted",
                                Port: nil,
                            },
                            ControllerName: "test.example.com",
                            Conditions: [
                                {
                                    Type: "Accepted",
                                    Status: "True",
                                    ObservedGeneration: 2,
                                    LastTransitionTime: {
                                        Time: 2022-11-04T09:44:40.528817-04:00,
                                    },
                                    Reason: "",
                                    Message: "",
                                },
                            ],
                        },
                        {
                            ParentRef: {
                                Group: nil,
                                Kind: nil,
                                Namespace: "test",
                                Name: "gateway",
                                SectionName: "not-accepted",
                                Port: nil,
                            },
                            ControllerName: "test.example.com",
                            Conditions: [
                                {
                                    Type: "Accepted",
                                    Status: "False",
                                    ObservedGeneration: 1,
                                    LastTransitionTime: {
                                        Time: 2022-11-04T09:44:40.528817-04:00,
                                    },
                                    Reason: "",
                                    Message: "",
                                },
                            ],
                        },
                    ],
                },
            }
--- FAIL: TestPrepareHTTPRouteStatus (0.00s)
```

But we can also do this:
```
	g.Expect(cmp.Diff(expected, result)).Should(gomega.BeEmpty(), "prepareHTTPRouteStatus() mismatch (-want +got)")
```

which will print:
```
=== RUN   TestPrepareHTTPRouteStatus
    httproute_test.go:92: 
        prepareHTTPRouteStatus() mismatch (-want +got)
        Expected
            <string>:   v1beta1.HTTPRouteStatus{
              	RouteStatus: v1beta1.RouteStatus{
              		Parents: []v1beta1.RouteParentStatus{
              			{
              				ParentRef:      {Namespace: &"test", Name: "gateway", SectionName: &"accepted"},
              				ControllerName: "test.example.com",
              				Conditions: []v1.Condition{
              					{
              						Type:               "Accepted",
              						Status:             "True",
            - 						ObservedGeneration: 2,
            + 						ObservedGeneration: 1,
              						LastTransitionTime: {Time: s"2022-11-04 09:47:36.07174 -0400 EDT m=+0.017945556"},
              						Reason:             "",
              						Message:            "",
              					},
              				},
              			},
              			{ParentRef: {Namespace: &"test", Name: "gateway", SectionName: &"not-accepted"}, ControllerName: "test.example.com", Conditions: {{Type: "Accepted", Status: "False", ObservedGeneration: 1, LastTransitionTime: {Time: s"2022-11-04 09:47:36.07174 -0400 EDT m=+0.017945556"}, ...}}},
              		},
              	},
              }
            
        to be empty
--- FAIL: TestPrepareHTTPRouteStatus (0.00s)

FAIL
```

let me know you thoughts.

however, I'm moving deduplication to the conditions package, so the comment no longer applies  (but it will become relevant in the future).

Reply:
I think https://github.com/nginxinc/nginx-kubernetes-gateway/blob/a67e7213b81d3f80de7de6515d54ed796b4eeb4e/internal/helpers/helpers.go#L12 is helpful in these cases. 

I prefer to start using gomega matching in all new unit tests. Eventually, it would be great to have consistency across all our unit tests.